dyno-queues-redis/src/main/java/com/netflix/dyno/queues/redis/RedisDynoQueue.java

```diff
@@ -160,8 +160,7 @@ public int getUnackTime() {
         Stopwatch sw = monitor.start(monitor.push, messages.size());
 
         try {
-
-            execute(() -> {
+            execute("push", "(a shard in) " + queueName, () -> {
```

Comment:
Split this into two parts: [shard, queue], pass empty value or null when one of them in not available.